DataStructureMemories/DataStructureMemories/DataStructureViewControllers/SFSafaryViewController.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  SFSafaryViewController.swift
+//  DataStructureMemories
+//
+//  Created by Alina Zaitseva on 4/23/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+
+import UIKit
+import SafariServices
+
+class SFSafaryViewController: UIViewController, SFSafariViewControllerDelegate {
+
+    var exactURL: String?
+    var isClosing = false
+   
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        guard let url = URL(string: self.exactURL!) else {
```

Comment:
You'll crash if `self.exactURL == nil`
If 'exactURL' is required, define it as `var exactURL: String!`

Reply:
Same for other controllers.